prompb/io/prometheus/client/decoder.go

```diff
@@ -236,8 +237,9 @@ func parseLabel(dAtA []byte, b scratchBuilder) error {
 				return io.ErrUnexpectedEOF
 			}
 			name = dAtA[iNdEx:postIndex]
-			if !model.LabelName(name).IsValid() {
-				return fmt.Errorf("invalid label name: %s", name)
+			unsafeName = yoloString(name)
```

Comment:
why unsafe here?

Reply:
Because `yoloString` returns an unsafe string.

Reply:
Sorry, I mean why is it unsafe to use name as-is when formatting the error string?

Reply:
`name` is a byte slice, you have to convert it to a string before using it as a string formatting argument, no? The `unsafe` part doesn't say anything about the `fmt.Errorf` call, but rather the string variable itself (`unsafeName`).